.travis.yml

```diff
@@ -0,0 +1,43 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+language: node_js
+node_js:
+  - "12"
+
+git:
+  depth: 1
+
+services:
+  - 'docker'
+
+before_script:
+  - npm install -g @angular/cli
+  - npm install karma
+  - npm install -g protractor
+  - npm install yarn
+
+jobs:
+  include:
+    - stage: pre-commit checks
+      script:
+        - make check-license
+        - make build-prod
+        - yarn test:coverage
+        - yarn lint
+      after_success: push image
```

Comment:
Shouldn't this be like `DOCKER_USERNAME=<username> DOCKER_PASSWORD=<password> REGISTRY=apache make push` ???

Reply:
@kingamarton 
The make push command expects atleast DOCKER_USERNAME and DOCKER_PASSWORD. If the REGISTRY is not yunikorn, then it also should be passed in. The default VERSION is set to latest, so version should be fine.

cc: @yangwwei  @wilfred-s  @sunilgovind

Reply:
That is how it is implemented in the shim side, but here, if we handle the push_image part in the same stage as the precommit checks it seems to not work.

Reply:
I will add the credentials in the Travis env for this and rerun the job.